guidelines/project-best-practices.md

```diff
@@ -0,0 +1,197 @@
+  - Uncheck branch protection rule ["Require branches to be up to date before merging"](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging) to reduce number of runs (potentially add a scheduled run if you are concerned about incompatible PRs getting merged)
+  - Use [filters](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) to eliminate unnecessary runs, e.g. doc PRs shouldn't require building and testing code.
+  - Consider running some jobs on schedule (nightly) rather than on each pull request (e.g. full matrix of platform tests)
+  - Inspect Github Actions run results on your own fork prior to opening Pull Request
+  - Contact Hyperledger staff for the possibility of dedicated paid runners
+* Pull request checks
+  - DCO - [Developer Certificate of Origin](https://github.com/apps/dco)
+  - Unit tests
+  - Integration tests
+  - Scans - see [Security](#Security) section, consider running on schedule (nightly) rather than on each pull request
+  - Note - just because a change passes automated checks doesn't mean it is necessarily good, it still requires judicious maintainer review
+* Test coverage reporting - run on-demand or nightly
+* Keep CI clean and green at all times, address failures and flakes
+* See also proposed [Automated Pipelines task force](https://github.com/hyperledger/toc/issues/44)
+
+## GitHub configuration
+
+* Define repository settings in `.github/settings.yml` so that they can be managed and tracked via pull requests, see [Fabric example](https://github.com/hyperledger/fabric/blob/main/.github/settings.yml).
+* Consider using a CODEOWNERS file to specify write permission per directory
+  - See [Fabric example](https://github.com/hyperledger/fabric/blob/main/CODEOWNERS) with additional `/docs` maintainers.
+  - Add a link from CODEOWNERS to MAINTAINERS.md scope field so that users can find domain area contacts
+* Consider using a .github/[PULL_REQUEST_TEMPLATE.md](https://raw.githubusercontent.com/hyperledger/fabric/main/.github/PULL_REQUEST_TEMPLATE.md) and .github/[ISSUE_TEMPLATE](https://github.com/hyperledger/fabric/tree/main/.github/ISSUE_TEMPLATE)
+* Define Branch protection rules. Recommended configuration:
+  - Check `Require a pull request before merging`
+  - Check `Require approvals`
+  - Check `Dismiss stale pull request approvals when new commits are pushed`
+  - Check `Require review from Code Owners` to enforce CODEOWNERS file.
+  - Check `Require status checks to pass before merging`
+  - Uncheck `Require branches to be up to date before merging` to limit CI runs
+  - Check `Restrict who can push to matching branches`
+  - The remainder can remain unchecked unless there is a specific need
```

Comment:
@ryjones Did I miss anything important in the branch protection rules?

Reply:
No, this is good